src/main/core/worker.rs

```diff
@@ -524,4 +529,47 @@ mod export {
         *global_alloc_counter = Counter::new();
         *global_dealloc_counter = Counter::new();
     }
+
+    /// Add the counters to their global counterparts, and clear the provided counters.
+    #[no_mangle]
+    pub extern "C" fn worker_addToGlobalAllocCounters(
+        alloc_counter: *mut Counter,
+        dealloc_counter: *mut Counter,
+    ) {
+        let alloc_counter = unsafe { alloc_counter.as_mut() }.unwrap();
+        let dealloc_counter = unsafe { dealloc_counter.as_mut() }.unwrap();
+
+        let mut global_alloc_counter = ALLOC_COUNTER.lock().unwrap();
+        let mut global_dealloc_counter = DEALLOC_COUNTER.lock().unwrap();
+
+        global_alloc_counter.add_counter(alloc_counter);
+        global_dealloc_counter.add_counter(dealloc_counter);
+
+        *alloc_counter = Counter::new();
+        *dealloc_counter = Counter::new();
+    }
+
+    /// Add the global counter to the provided counter, and clear the global counter.
+    #[no_mangle]
+    pub extern "C" fn worker_addFromGlobalSyscallCounter(
+        syscall_counter: *mut Counter,
+    ) {
+        let syscall_counter = unsafe { syscall_counter.as_mut() }.unwrap();
+
+        let mut global_syscall_counter = SYSCALL_COUNTER.lock().unwrap();
+        syscall_counter.add_counter(&global_syscall_counter);
```

Comment:
Nit - these would be a little clearer if we added something like `Counter::take(&mut self)` that returns the current current count and resets to 0.